### PR TITLE
Ability Toggles & Speed Calcs Upgrade - 3/14/19

### PR DESCRIPTION
**Pull request by WheatiesTM**
I've added the following features to the Damage Calc:
- calculating and displaying the final speed stat in all gens
- a toggle for the Battery ability
- a toggle for the Tailwind field effect
- a hide-able toggle that controls abilities with odd activation conditions

### CODE REVIEW

**Comment by AustinXII** [comment]
Can you fix the Travis error?

**Comment by WheatiesTM** [comment]
> Can you fix the Travis errors?

Yeah I'll work on it

**Comment by AustinXII** [comment]
Sweet! Thanks ill t at this later and merge afterwards unless you have a screenshots

**Comment by WheatiesTM** [comment]
Don't have screenshots (not sure what you mean by that tho, is it common to include actual screenshots of the result of the code?)

**Comment by AustinXII** [comment]
Yes, this has a UI change and I’m very curious what this looks like now that we have a lot more buttons.

**Comment by WheatiesTM** [comment]
damage calc UI

**Comment by WheatiesTM** [comment]
sorry didn't know that "Close and comment" meant close the pull request lol

File: js/shared_controls.js [hunk 1]
```diff
@@ -573,6 +587,7 @@ function Pokemon(pokeInfo) {
 		}
 		this.nature = pokeInfo.find(".nature").val();
 		this.ability = pokeInfo.find(".ability").val();
+		this.abilityOn = 	pokeInfo.find(".abilityToggle").is(":checked");
```
Comment (AustinXII) [inline js/shared_controls.js]: Same here

File: index.template.html [hunk 2]
```diff
@@ -382,6 +382,8 @@
                                     <option value="-6">-6</option>
                                 </select>
                             </td>
+                            <td><span class="totalMod">---</span>
```
Comment (AustinXII) [inline index.template.html]: Are these dashes needed?

File: honkalculate.template.html [hunk 3]
```diff
@@ -334,6 +334,8 @@
                                         <option value="-6">-6</option>
                                     </select>
                                 </td>
+                                <td><span class="totalMod">---</span>
```
Comment (AustinXII) [inline honkalculate.template.html]: Same with these dashes

File: js/damage.js [hunk 4]
```diff
@@ -41,17 +43,18 @@ function CALCULATE_MOVES_OF_ATTACKER_BW(attacker, defender, field) {
...
 	defender.stats[SD] = getModifiedStat(defender.rawStats[SD], defender.boosts[SD]);
-	defender.stats[SP] = getFinalSpeed(defender, field.getWeather());
+	defender.stats[SP] = getFinalSpeed(defender, field, defenderSide);
 	checkIntimidate(attacker, defender);
 	checkIntimidate(defender, attacker);
 	checkDownload(attacker, defender);
 	attacker.stats[AT] = getModifiedStat(attacker.rawStats[AT], attacker.boosts[AT]);
 	attacker.stats[SA] = getModifiedStat(attacker.rawStats[SA], attacker.boosts[SA]);
 	defender.stats[AT] = getModifiedStat(defender.rawStats[AT], defender.boosts[AT]);
-	var defenderSide = field.getSide(~~(mode === "one-vs-all"));
```
Comment (AustinXII) [inline js/damage.js]: Intentional?

File: index.template.html [hunk 5]
```diff
@@ -1139,6 +1166,7 @@
                 <div class="gen-specific g3 g4 g5 g6 g7">
                     <label for="abilityR1">Ability</label>
                     <select class="ability terrain-trigger calc-trigger" id="abilityR1"></select>
+                    <input hidden type="checkbox" class="abilityToggle calc-trigger" checked>
```
Comment (AustinXII) [inline index.template.html]: We should probably have a title to explain what that checkbox is doing

File: js/shared_controls.js [hunk 6]
```diff
@@ -529,6 +541,7 @@ function Pokemon(pokeInfo) {
 		}
 		this.ability = (set.ability && typeof set.ability !== "undefined") ? set.ability :
 			(pokemon.ab && typeof pokemon.ab !== "undefined") ? pokemon.ab : "";
+		this.abilityOn = 	true;
```
Comment (AustinXII) [inline js/shared_controls.js]: Too many spaces

**Comment by AustinXII** [comment]
Don’t worry lol, that happens to everyone at least once

File: index.template.html (hunk 2 above)
Comment (WheatiesTM) [reply index.template.html]: the dashes are just a placeholder to show that something should be there while it's calculating, not strictly necessary but there were numbers manually written in for the other stats so I thought something should go there

File: js/damage.js (hunk 4 above)
Comment (WheatiesTM) [reply js/damage.js]: yes, i just moved that line up a few so that the speed calcs could access the side

File: index.template.html (hunk 5 above)
Comment (WheatiesTM) [reply index.template.html]: yeah good idea

File: js/shared_controls.js (hunk 6 above)
Comment (WheatiesTM) [reply js/shared_controls.js]: right, that was a tab i think which just looked like an extra space to me

**Comment by AustinXII** [comment]
Awesome looks good! Thanks :)

**Pull request merged**